internal/controller/streams_helper.go

```diff
@@ -149,24 +149,40 @@ func (sm *streamManager) sendKeepalive(logger *zap.Logger, st StreamType) error
 
 	switch st {
 	case STREAM_NETWORK_FLOWS:
+		if sm.streamClient.networkFlowsStream == nil {
+			return nil // If the stream is not initialized, we don't need to send a keepalive.
+		}
+
 		err = sm.streamClient.networkFlowsStream.Send(&pb.SendKubernetesNetworkFlowsRequest{
 			Request: &pb.SendKubernetesNetworkFlowsRequest_Keepalive{
 				Keepalive: &pb.Keepalive{},
 			},
 		})
 	case STREAM_RESOURCES:
+		if sm.streamClient.resourceStream == nil {
```

Comment:
Following the code, I don't see how this field could ever be `nil` at runtime.
Revert this.